pkg/ccl/backupccl/backup_test.go

```diff
@@ -8703,3 +8703,25 @@ DROP TABLE foo;
 	defer cleanupEmptyCluster()
 	sqlDBRestore.Exec(t, "RESTORE FROM $1 AS OF SYSTEM TIME "+aost, LocalFoo)
 }
+
+func TestRestoreNewDatabaseName(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	const numAccounts = 1
+	_, _, sqlDB, _, cleanupFn := BackupRestoreTestSetup(t, singleNode, numAccounts, InitManualReplication)
+	defer cleanupFn()
+
+	sqlDB.Exec(t, `BACKUP DATABASE data TO $1`, LocalFoo)
+
+	// Should fail because 'data' database is still in cluster
+	sqlDB.ExpectErr(t, "database",
+		"RESTORE DATABASE data FROM $1", LocalFoo)
+
+	// Should pass because 'new_data' is not in cluster
+	sqlDB.Exec(t, "RESTORE DATABASE data FROM $1 WITH new_db_name = 'new_data'", LocalFoo)
```

Comment:
Once we restore, we should check that the database has actually been created. We can do this by creating a table and verifying rows have been restored, as mentioned in the comment above. We could also do this by checking that an entry with the name `new_data` can be seen in the output of `SHOW DATABASES`.

Reply:
done